zerver/actions/user_settings.py

```diff
@@ -63,12 +99,19 @@ def do_change_user_delivery_email(user_profile: UserProfile, new_email: str) ->
     else:
         user_profile.save(update_fields=["delivery_email"])
 
-    # We notify just the target user (and eventually org admins, only
-    # when email_address_visibility=EMAIL_ADDRESS_VISIBILITY_ADMINS)
-    # about their new delivery email, since that field is private.
+    # We notify all the users who have access to delivery email.
     payload = dict(user_id=user_profile.id, delivery_email=new_email)
     event = dict(type="realm_user", op="update", person=payload)
-    transaction.on_commit(lambda: send_event(user_profile.realm, event, [user_profile.id]))
+    active_users = user_profile.realm.get_active_users()
+    delivery_email_visible_user_ids = [
+        user.id
+        for user in active_users
```

Comment:
Something like `for active_user in active_users` might be a nicer variable name -- I don't like ever having two variables `user_profile` and `user` in the same function. Eventually we're probably going to rename `UserProfile` to `User` and try to squash all of those.

Reply:
Updated this.